lib/src/main/java/com/thenewboston/config/datasource/ConfigDataSource.kt

```diff
@@ -0,0 +1,23 @@
+package com.thenewboston.config.datasource
+
+import com.thenewboston.common.http.NetworkClient
+import com.thenewboston.common.http.Outcome
+import com.thenewboston.common.http.makeApiCall
+import com.thenewboston.data.dto.bankapi.configdto.ConfigDTO
+import io.ktor.client.request.*
+
+private const val CONFIG_ENDPOINT = "/config"
+
+class ConfigDataSource(private val networkClient: NetworkClient) {
+    suspend fun fetchConfig(): Outcome<ConfigDTO> = makeApiCall(
+        call = { doFetchConfig() },
+        errorMessage = "Could not fetch config"
+    )
+
+    private suspend fun doFetchConfig(): Outcome<ConfigDTO> {
+        val urlSuffix = "$CONFIG_ENDPOINT"
+        val config = networkClient.client.get<ConfigDTO>(urlSuffix)
+
+        return Outcome.Success(config)
```

Comment:
We can't just return a success result every time. If the API request fails, we will need to handle that. 

```
`val urlSuffix = "$CONFIG_ENDPOINT"
        val config = networkClient.client.get<ConfigDTO>(urlSuffix)
        
        return when {
            config.account_number.isNullOrEmpty() -> Outcome.Error("Received null or empty list", IOException())
            else -> Outcome.Success(config)
        }
```

Reply:
Yes agree, thanks